test/powershell/Modules/Microsoft.PowerShell.Core/Import-Module.Tests.ps1

```diff
@@ -22,3 +22,52 @@
         (Get-Module -Name $moduleName).Name | Should Be $moduleName
     }
 }
+
+Describe "Import-Module with ScriptsToProcess" -Tags "CI" {
+    
+    BeforeAll {
+        $moduleRootPath = Join-Path $TestDrive 'TestModules'
+        New-Item $moduleRootPath -ItemType Directory -Force | Out-Null
+        Push-Location $moduleRootPath
+
+        "1 | Out-File out.txt -Append -NoNewline" | Out-File script1.ps1
+        "2 | Out-File out.txt -Append -NoNewline" | Out-File script2.ps1
+        New-ModuleManifest module1.psd1 -ScriptsToProcess script1.ps1
+        New-ModuleManifest module2.psd1 -ScriptsToProcess script2.ps1 -NestedModules module1.psd1
+    }
+
+    AfterAll {
+        Pop-Location
+        #Remove-Item $moduleRootPath -Recurse -Force -ErrorAction SilentlyContinue
+    }
+
+    BeforeEach {
+        New-Item out.txt -ItemType File -Force | Out-Null
+    }
+
+    AfterEach {
+        $m = @('module1','module2','script1','script2')
+        remove-module $m -Force -ErrorAction SilentlyContinue
+        Remove-Item out.txt -Force -ErrorAction SilentlyContinue
+    }
+    
+    It "Verify ScriptsToProcess are executed for top-level module" {
```

Comment:
You can have custom test case names for individual test cases. Have look at: https://github.com/PowerShell/PowerShell/blob/02b5f357a20e6dee9f8e60e3adb9025be3c94490/test/powershell/engine/ParameterBinding/NullableBooleanDCR.Tests.ps1

Here the parameters will be parameter splat for Import-Module, and expected result, test name.